packages/start/api/internalFetch.ts

```diff
@@ -8,16 +9,67 @@ export const registerApiRoutes = routes => {
   apiRoutes = routes;
 };
 
-export async function internalFetch(route: string, init: RequestInit) {
+// Headers from the original page GET request that should not
+// be copied over to the fetch request
+const SKIPPED_HEADERS = ["Accept"];
+
+// https://fetch.spec.whatwg.org/#forbidden-header-name
+const FORBIDDEN_HEADERS = [
+  `Accept-Charset`,
+  `Accept-Encoding`,
+  `Access-Control-Request-Headers`,
+  `Access-Control-Request-Method`,
+  `Connection`,
+  `Content-Length`,
+  `Cookie`,
+  `Cookie2`,
+  `Date`,
+  `DNT`,
+  `Expect`,
+  `Host`,
+  `Keep-Alive`,
+  `Origin`,
+  `Referer`,
+  `Set-Cookie`,
+  `TE`,
+  `Trailer`,
+  `Transfer-Encoding`,
+  `Upgrade`,
+  `Via`
+];
+
+export async function internalFetch(route: string, init: RequestInit = {}) {
   if (route.startsWith("http")) {
     return await fetch(route, init);
   }
 
-  let url = new URL(route, "http://internal");
-  const request = new Request(url.href, init);
+  const headers = new Headers(useServerContext().request?.headers);
```

Comment:
this `useServerContext()` is probably not going to work correctly. I want to solve this a different way. We need `server.$fetch` to actually go through the whole handler pipeline so people from an API routes can request a UI page and kind of pretend to have rewritten that url.. maybe we can merge and we can build on top... but basically we need an internal fetch that closes over the current call to fetch, like have a reference to the `handler` so that we can run an internal fetch through the whole handler, not just the API routes

Reply:
That sounds good to me, do you have any idea how else we could access the headers other than `useServerContext()`?